cse-x345-fall-2017/05-coders-choice
# 05 coders choice final submit

1) Dependency ecto and postgres

mix ecto.create
mix ecto.migrate

2) Uses product, Cart and UserService APIs in separate supervised application "Shopservices"

3) Testing
	
	MY MVC appl with CRUD is in place (see /shopping folder - mix phx.server)
	
	Creating mostly using ...
	mix phoenix.gen.html Product products 
	mix phoenix.gen.html Cart carts 

	http://localhost:4000/carts
	http://localhost:4000/products

	Its not completly wired up and I was running out of time, so I tested from iex (see below)

4) Add products

	product1 = Shopservices.ProductService.create("A12345Z", "Cool product one", 1, "ea", 5.00, "/images/cproduct1.png")

	product2 = Shopservices.ProductService.create("A12346Y", "Cool product two", 1, "ea", 6.00, "/images/cproduct2.png")

4) Create cart items (this part i was still working out  so for testing I got the Id's from the added products)

	cart_items = [%{product_id: product1.id, price: 2.5, quanity: 2}, %{product_id: product2.id, price: 20, quanity: 1}]

5) Add items to cart for this user

	Shopservices.CartService.create(%{customer: "Joe Davault", date: Ecto.Date.utc, cart_items: cart_items})
